CMakeLists.txt

```diff
@@ -0,0 +1,97 @@
+endif()
+
+set(COMPILEOPTION_DATAFILEPATH ${CMAKE_INSTALL_PREFIX} CACHE PATH "Set the folder where will be the datas; this is CMAKE_INSTALL_PREFIX" FORCE)
+set(COMPILEOPTION_SAVEFOLDERNAME "Blackvoxel" CACHE PATH "Folder where datas, universes... are stocked")
+set(COMPILEOPTION_USEHOMEDIRSTORAGE ON CACHE BOOL "Is save folder in HOME?")
+
+set(GAME_SOURCE_FILE "")
+file(GLOB GAME_SOURCE_FILE src/*.cpp)
+set(GAME_SOURCE_FILE_TOADD "")
+file(GLOB GAME_SOURCE_FILE_TOADD src/z/*.cpp)
+set(GAME_SOURCE_FILE ${GAME_SOURCE_FILE} ${GAME_SOURCE_FILE_TOADD})
+unset(GAME_SOURCE_FILE_TOADD)
+
+set(addto_CMAKE_FLAGS "-Isrc/sc_Squirrel3/include")
+
+if(WIN32)
+	set(addto_CMAKE_FLAGS "${addto_CMAKE_FLAGS} -O3 -fmessage-length=0 -march=i686")
+	set(setto_CMAKE_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS} -s -Xlinker --large-address-aware -mwindows -L\"src/sc_Squirrel3/lib\" -lmingw32 -lSDLmain -lSDL -llibglew32 -lglu32 -lopengl32 -llibglut -lsquirrel -lsqstdlib")
+else(WIN32)
+	if("${CMAKE_SYSTEM_NAME}" MATCHES "Linux")
+		set(addto_CMAKE_FLAGS "${addto_CMAKE_FLAGS} -O3 -fmessage-length=0")
+		set(setto_CMAKE_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS} -s -zrelro -L\"src/sc_Squirrel3/lib\" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib")
+	elseif("${CMAKE_SYSTEM_NAME}" MATCHES "FreeBSD")
+		set(addto_CMAKE_FLAGS "${addto_CMAKE_FLAGS} -O3 -fmessage-length=0")
+		set(setto_CMAKE_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS} -s -L\"src/sc_Squirrel3/lib\" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib")
+	elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Darwin")
+		set(addto_CMAKE_FLAGS "${addto_CMAKE_FLAGS} -O3 -fmessage-length=0")
+		set(setto_CMAKE_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS} -s -L\"src/sc_Squirrel3/lib\" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib")
+	else()
+		message(AUTHOR_WARNING "The system (${CMAKE_SYSTEM_NAME}) isn't recognized, using default config.")
+		set(addto_CMAKE_FLAGS "${addto_CMAKE_FLAGS} -O3 -fmessage-length=0")
+		set(setto_CMAKE_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS} -s -L\"src/sc_Squirrel3/lib\" -lGLU -lSDL -lGLEW -lGL -lsquirrel -lsqstdlib")
+	endif()
+endif(WIN32)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${addto_CMAKE_FLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} ${addto_CMAKE_FLAGS}")
+
+unset(addto_CMAKE_FLAGS)
+
+set(CMAKE_EXE_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS}")
+set(CMAKE_MODULE_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS}")
+set(CMAKE_SHARED_LINKER_FLAGS "${setto_CMAKE_LINKER_FLAGS}")
+
+unset(setto_CMAKE_LINKER_FLAGS)
+
+add_subdirectory(src/sc_Squirrel3)
+
+set(EXTRA_LIBS squirrel sqstdlib)
+
+add_executable(blackvoxel ${GAME_SOURCE_FILE})
+target_link_libraries(blackvoxel ${EXTRA_LIBS})
+set_target_properties(blackvoxel PROPERTIES COMPILE_DEFINITIONS "COMPILEOPTION_DATAFILESPATH=\"${CMAKE_INSTALL_PREFIX}\";COMPILEOPTION_SAVEFOLDERNAME=\"${COMPILEOPTION_SAVEFOLDERNAME}\";COMPILEOPTION_USEHOMEDIRSTORAGE=\"${COMPILEOPTION_USEHOMEDIRSTORAGE}\"")
```

Comment:
I just see that `\"` is broken in some Make version (see https://cmake.org/cmake/help/v3.7/prop_tgt/COMPILE_DEFINITIONS.html)
The use of a Makefile is strongly recommanded then if you are in this case... (Hopefully you can always do `git add Makefile` or `git checkout Makefile`!)